internal/controller/api/job_receiver.go

```diff
@@ -1,23 +1,54 @@
 package api
 
 import (
+	"fmt"
+	"io"
+
 	"github.com/gorilla/mux"
 
 	//	"context"
 	"encoding/json"
-	"io"
-	"io/ioutil"
 	"log"
 	"net/http"
 
 	"github.com/google/uuid"
 
 	"github.com/RedHatInsights/platform-receptor-controller/internal/controller"
+	"github.com/go-playground/validator/v10"
 	"github.com/redhatinsights/platform-go-middlewares/identity"
 
 	kafka "github.com/segmentio/kafka-go"
 )
 
+type badRequest struct {
+	status int
+	msg    string
+}
+
+func (br *badRequest) Error() string {
+	return fmt.Sprintf("%d: %s", br.status, br.msg)
+}
+
+func decodeJSON(body io.ReadCloser, job interface{}) error {
+	dec := json.NewDecoder(body)
+	if err := dec.Decode(&job); err != nil {
+		// FIXME: More specific error handling needed.. case statement for different scenarios?
+		return &badRequest{status: http.StatusBadRequest, msg: "Request body includes malformed json"}
```

Comment:
For now, what do you think about removing badRequest{} instance and doing something like this:

`
  return errors.New("Request body includes malformed json")
`

Reply:
I guess now we aren't really gaining anything from it, but eventually if we have different error statuses it would be convenient to pass that along so that the handler could just write the error it was given. If StatusBadRequest is the only error status we'll be returning then it'd make sense to get rid of badRequest{}. Wasn't sure if that was the case or not

Reply:
I was thinking we could make the decodeJSON() method to be as generic as possible.  It should just be responsible for handling the json decoding and validation (maybe that's even too much as it seems like it will have 2 jobs...but that feels okay for now).  The method could still return different types of errors and the caller can determine how to convert those errors into what the caller needs to return.  Plus I think this makes it possible to reuse this method outside of the context of an http call (not sure that will happen, but its always possible...and trying to increase reuse always sounds good in theory  :) ).

I hope that makes sense.  Let me know if we need to discuss.

Reply:
Yeah works for me! It might make sense to eventually pull out the decoding and validation into separate utilities